tavern/app.go

```diff
@@ -107,54 +111,73 @@ func NewServer(ctx context.Context, options ...func(*Config)) (*Server, error) {
+	} else {
+		withAuthentication = tavernhttp.WithAuthenticationBypass(client)
+	}
+
+	// Configure Request Logging
 	httpLogger := log.New(os.Stderr, "[HTTP] ", log.Flags())
-	router := http.NewServeMux()
-	router.Handle("/status", newStatusHandler())
-	router.Handle("/oauth/login", auth.NewOAuthLoginHandler(cfg.oauth, privKey))
-	router.Handle("/oauth/authorize", auth.NewOAuthAuthorizationHandler(cfg.oauth, pubKey, client, "https://www.googleapis.com/oauth2/v3/userinfo"))
-	router.Handle("/graphql", newGraphQLHandler(client))
-	router.Handle("/cdn/", cdn.NewDownloadHandler(client))
-	router.Handle("/cdn/upload", cdn.NewUploadHandler(client))
-	router.Handle("/", auth.WithLoginRedirect("/oauth/login", www.NewHandler(httpLogger)))
-	router.Handle("/playground", auth.WithLoginRedirect("/oauth/login", playground.Handler("Tavern", "/graphql")))
+
+	// Route Map
+	routes := tavernhttp.RouteMap{
+		"/status":      tavernhttp.Endpoint{Handler: newStatusHandler()},
+		"/oauth/login": tavernhttp.Endpoint{Handler: auth.NewOAuthLoginHandler(cfg.oauth, privKey)},
+		"/oauth/authorize": tavernhttp.Endpoint{Handler: auth.NewOAuthAuthorizationHandler(
+			cfg.oauth,
+			pubKey,
+			client,
+			"https://www.googleapis.com/oauth2/v3/userinfo",
+		)},
+		"/graphql":    tavernhttp.Endpoint{Handler: newGraphQLHandler(client)},
+		"/c2.C2/":     tavernhttp.Endpoint{Handler: newGRPCHandler(client)},
+		"/grpc/":      tavernhttp.Endpoint{Handler: newGRPCHandler(client)},
+		"/cdn/":       tavernhttp.Endpoint{Handler: cdn.NewDownloadHandler(client)},
+		"/cdn/upload": tavernhttp.Endpoint{Handler: cdn.NewUploadHandler(client)},
+		"/": tavernhttp.Endpoint{
+			Handler:          www.NewHandler(httpLogger),
+			LoginRedirectURI: "/oauth/login",
+		},
+		"/playground": tavernhttp.Endpoint{
+			Handler:          playground.Handler("Tavern", "/graphql"),
+			LoginRedirectURI: "/oauth/login",
+		},
+	}
 
 	// Setup Profiling
 	if cfg.IsPProfEnabled() {
 		log.Printf("[WARN] Performance profiling is enabled, do not use in production as this may leak sensitive information")
-		registerProfiler(router)
+		registerProfiler(routes)
 	}
 
-	// Log Middleware
-	handlerWithLogging := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		authName := "unknown"
-		id := auth.IdentityFromContext(r.Context())
-		if id != nil {
-			authName = id.String()
-		}
-
-		httpLogger.Printf("%s (%s) %s %s\n", r.RemoteAddr, authName, r.Method, r.URL)
-		router.ServeHTTP(w, r)
-	})
+	// Create Tavern HTTP Server
+	srv := tavernhttp.NewServer(
+		routes,
+		withAuthentication,
+		tavernhttp.WithRequestLogging(httpLogger),
+	)
 
-	// Auth Middleware
-	var endpoint http.Handler
-	if cfg.oauth.ClientID != "" {
-		endpoint = auth.Middleware(handlerWithLogging, client)
-	} else {
-		endpoint = auth.AuthDisabledMiddleware(handlerWithLogging, client)
-	}
+	// Configure HTTP/2 (support for without TLS)
+	handler := h2c.NewHandler(srv, &http2.Server{})
```

Comment:
Does GCP manage the TLS for our grpc endpoint? 🤯

Reply:
Yes